### ISSUE TITLE
keras ConvLSTM2d not working with eager execution

### ISSUE THREAD

**Issue by SimplifyAll** [tensorflow/tensorflow#25895]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: NA
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 1.13.0-dev20190130
- Python version: 3.6
- Bazel version (if compiling from source): NA
- GCC/Compiler version (if compiling from source): NA
- CUDA/cuDNN version: 10/7.4.2.24
- GPU model and memory: GeForce GTX 1050 Ti, 4096 MB

**Describe the current behavior**
While using Eager Execution keras ConvLSTM2D seems like not initialized.
After that we tried implement our model using gradient tape and it seemed fine but after loading checkpoint results was different than expected.
The second issue probably has something in common with the first, so i'm posting code to reproduce first one and logs. 

**Describe the expected behavior**
ConvLSTM2D works with Eager Execution

**Code to reproduce the issue**
```python
DATA_PATH = 'path/to/directory/with/grayscale/jpg/images"

import os
import tensorflow as tf
from tensorflow.keras import layers, Sequential

tf.enable_eager_execution()
filenames = [os.path.join(DATA_PATH, x) for x in sorted(os.listdir(DATA_PATH))]
tf_dataset = tf.data.Dataset.from_tensor_slices(filenames).map(lambda x: tf.image.decode_jpeg(tf.read_file(x), channels=0))
tf_dataset = tf_dataset.batch(10).batch(10)
tf_dataset = tf.data.Dataset.zip((tf_dataset, tf_dataset)).prefetch(tf.contrib.data.AUTOTUNE)

model =  Sequential([
    layers.ConvLSTM2D(filters=3, kernel_size=3, padding='same')
])
model.compile(optimizer=tf.train.AdamOptimizer(), loss='categorical_crossentropy', metrics=['accuracy'])
model.fit(tf_dataset)
```

**Other info / logs**
```
WARNING: Logging before flag parsing goes to stderr.
W0219 14:04:10.508448 140324682260608 training_utils.py:1249] Expected a shuffled dataset but input dataset `x` is not shuffled. Please invoke `shuffle()` on input dataset.

2019-02-19 14:04:13.426855: W tensorflow/core/common_runtime/base_collective_executor.cc:203] BaseCollectiveExecutor::StartAbort Failed precondition: Error while reading resource variable _AnonymousVar4 from Container: localhost. This could mean that the variable was uninitialized. Not found: Resource localhost/_AnonymousVar4/N10tensorflow3VarE does not exist.
         [[{{node convlstm1/convolution/ReadVariableOp}}]]
         [[training/TFOptimizer/gradients/convlstm1/while_grad/convlstm1/while_grad/LoopCond/_705/_52]]

2019-02-19 14:04:13.426894: W tensorflow/core/common_runtime/base_collective_executor.cc:203] BaseCollectiveExecutor::StartAbort Failed precondition: Error while reading resource variable _AnonymousVar4 from Container: localhost. This could mean that the variable was uninitialized. Not found: Resource localhost/_AnonymousVar4/N10tensorflow3VarE does not exist.
         [[{{node convlstm1/convolution/ReadVariableOp}}]]

2019-02-19 14:04:13.426968: E tensorflow/core/common_runtime/process_function_library_runtime.cc:735] Component function execution failed: Failed precondition: Error while reading resource variable _AnonymousVar4 from Container: localhost. This could mean that the variable was uninitialized. Not found: Resource localhost/_AnonymousVar4/N10tensorflow3VarE does not exist.
         [[{{node convlstm1/convolution/ReadVariableOp}}]]
         [[training/TFOptimizer/gradients/convlstm1/while_grad/convlstm1/while_grad/LoopCond/_705/_52]]

2019-02-19 14:04:13.427323: E tensorflow/core/common_runtime/process_function_library_runtime.cc:735] Component function execution failed: Failed precondition: Error while reading resource variable _AnonymousVar4 from Container: localhost. This could mean that the variable was uninitialized. Not found: Resource localhost/_AnonymousVar4/N10tensorflow3VarE does not exist.
         [[{{node convlstm1/convolution/ReadVariableOp}}]]
```

**Comment by liuqiang3** [comment]
I encountered the same preblem in tensorflow-gpu-1.9

**Comment by qlzh727** [comment]
I think the issue should be the fixed by the recent commit https://github.com/tensorflow/tensorflow/commit/391bee73641f1c23ef90cc40a9e8893f343c30fc. Can u try the same snippet with tf-nightly and see if it works?

**Comment by liuqiang3** [comment]
> I think the issue should be the fixed by the recent commit [391bee7](https://github.com/tensorflow/tensorflow/commit/391bee73641f1c23ef90cc40a9e8893f343c30fc). Can u try the same snippet with tf-nightly and see if it works?
I install  tf-nightly -1.14.1-dev20190304, I re-run the code to find it started wording.
thank you!

**Comment by liuqiang3** [comment]
I install tf-nightly -1.14.1-dev20190304, I re-run the code to find it started wording.
thank you!

**Comment by qlzh727** [comment]
SG, glad its fixed.
